components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -803,4 +810,288 @@ public void testUpdateUserWithUsernameChange() throws Exception {
+            add(new LocalClaim("sample/organization", new ArrayList<AttributeMapping>(),
+                    notSupportedByDefaultProperties));
+            add(new LocalClaim("sample/manager", new ArrayList<AttributeMapping>(), properties));
+        }};
+        List<ExternalClaim> externalClaimMap = new ArrayList<ExternalClaim>() {{
+            add(new ExternalClaim(externalClaimURI, externalClaimURI + ":department",
+                    "sample/department"));
+            add(new ExternalClaim(externalClaimURI, externalClaimURI + ":organization",
+                    "sample/organization"));
+            add(new ExternalClaim(externalClaimURI, externalClaimURI + ":manager",
+                    "sample/manager"));
+        }};
+
+        when(mockClaimMetadataManagementService.getExternalClaims(SCIMCommonConstants.
+                SCIM_USER_CLAIM_DIALECT, tenantDomain)).thenReturn(externalClaimMap);
+        when(mockClaimMetadataManagementService.getLocalClaims(tenantDomain)).thenReturn(localClaimMap);
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager,
+                mockClaimMetadataManagementService, tenantDomain);
+        List<Attribute> list = scimUserManager.getUserSchema();
+        assertEquals(list.size(), 2);
+    }
+
+    @Test(dataProvider = "groupPermission")
+    public void testGetGroupPermissions(String roleName, String[] permission, Object expected) throws Exception {
+
+        String tenantDomain = "carbon.super";
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager,
+                mockClaimMetadataManagementService, tenantDomain);
+        mockStatic(SCIMCommonComponentHolder.class);
+        when(SCIMCommonComponentHolder.getRolePermissionManagementService())
+                .thenReturn(mockedRolePermissionManagementService);
+        when(mockedRolePermissionManagementService.getRolePermissions(anyString(), anyInt())).thenReturn(permission);
+        String[] actual = scimUserManager.getGroupPermissions(roleName);
+        assertEquals(expected, actual);
+    }
+
+    @DataProvider(name = "groupPermission")
+    public Object[][] groupPermission() throws Exception {
+
+        Group group = new Group();
+        group.setDisplayName("roleName");
+        String[] permission1 = new String[]{};
+        String[] permission2 = new String[]{"/permission/admin/login"};
+        return new Object[][]{
+                {null, permission1, permission1},
+                {"roleName", permission2, permission2}
+        };
+    }
+
+    @DataProvider(name = "getUserConfigurations")
+    public Object[][] getUserConfigurations() {
+
+        String username = "user";
+        String domainQualifiedUserName = "domainQualifiedUserName";
+        Map<String, String> userClaimValues1 = new HashMap<>();
+        userClaimValues1.put("http://wso2.org/claims/username" , username);
```

Comment:
Let's declare constants for duplicated strings like claim uris

Reply:
Addressed with commit [4772a01](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/347/commits/4772a0151ab531a20c6db0026ba7db58f9385445)

Reply:
You have introduced variables rather than constants.

There some more strings eligible for constants such as "ReadOnly", "SupportedByDefault" etc.

Reply:
Addressed with [5389dda](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/347/commits/5389ddafedb7b91ea10f171863a9e9a79913b178)